runner.go

```diff
@@ -85,25 +85,36 @@ func (r *Runner) getService(name, namespace string) (*v1.Service, error) {
 	)
 }
 
-func (r *Runner) createService(name, namespace string, labels map[string]string, ports []v1.ServicePort) (*v1.Service, error) {
-	// Always create clusterIP type (default type) services. There is no
-	// reason to create anything else for the purpose of mirroring
-	return r.client.CoreV1().Services(r.namespace).Create(
-		&v1.Service{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      name,
-				Namespace: namespace,
-				Labels:    labels,
-			},
-			Spec: v1.ServiceSpec{
-				Ports:    ports,
-				Selector: nil,
-			},
-		})
+func isHeadless(svc *v1.Service) bool {
```

Comment:
I don't understand this logic.

So you are checking `svc.Spec.ClusterIP == "None"`, and then on L110 - you are setting it to the same value: `svc.Spec.ClusterIP = "None"`.

Why do we need this code at all?

Reply:
In general when creating a service we do not set the `clusterIP` ourselves but let kube assign one from the available range. So for a service in gcp, let's say `10.5.0.100` we create a service without specifying the clusterIP and kube gives one to our mirrored service, for example `10.3.0.200` for mirroring the above gcp service in aws.
Headless services are a special case since you need to specify `clusterIP: "None"`, in this case you instruct kube to not assign a vip to the svc and the dns resolves in multiple A records for the svc endpoints, so if you want to replicate that behaviour and since we are not doing cross cluster dns anymore you need a local headless service with the same endpoints